### ISSUE TITLE
Odd bash syntax error not detected

### ISSUE THREAD

**Issue by memotype** [koalaman/shellcheck#1815]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1083 (kind of...)
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

This is a syntax error in bash (all versions I checked, up to 5.0.7(1)-release. Freenode #bash folks confirms it's a known bug that isn't likely to be fixed (they also say it's the same problem in zsh, but not in dash or ksh). 

```sh
#!/bin/bash
f() { 
  case "$c" in
    }) : ;;
  esac
}
```

However, if the `})` isn't on a new line, bash doesn't seem to mind:

```sh
#!/bin/bash
f() {
  case "$c" in })
    : ;;
  esac
}
```

#### Here's what shellcheck currently says:

Shellcheck merely warns "^-- SC1083: This } is literal. Check expression (missing ;/\n?) or quote it." in both cases.

#### Here's what I wanted or expected to see:

An error saying this is a syntax error if the shell is bash (or zsh, apparently).
